tmt/tests/booted/test-upgrade-download-only.nu

```diff
@@ -0,0 +1,117 @@
+# number: 30
+# extra:
+#   try_bind_storage: true
+# tmt:
+#   summary: Execute upgrade --download-only tests
+#   duration: 30m
+#
+# This test does:
+# bootc image copy-to-storage
+# podman build <from that image>
+# bootc upgrade --download-only
+# reboot (should boot into old image)
+# bootc upgrade --download-only (verify lock status)
+# bootc upgrade
+# reboot (should boot into new image)
+#
+use std assert
+use tap.nu
+
+# This code runs on *each* boot.
+# Here we just capture information.
+bootc status
+journalctl --list-boots
+
+let st = bootc status --json | from json
+let booted = $st.status.booted.image
+
+# Parse the kernel commandline into a list.
+# This is not a proper parser, but good enough
+# for what we need here.
+def parse_cmdline []  {
+    open /proc/cmdline | str trim | split row " "
+}
+
+def imgsrc [] {
+    $env.BOOTC_upgrade_image? | default "localhost/bootc-derived-local"
+}
+
+# Run on the first boot
+def initial_build [] {
+    tap begin "upgrade --download-only test"
+
+    let imgsrc = imgsrc
+    # For the packit case, we build locally right now
+    if ($imgsrc | str ends-with "-local") {
+        bootc image copy-to-storage
+
+        # A simple derived container that adds a file
```

Comment:
I'm not sure we need another copy of this. How about integrating `--download-only` alongside say `test-image-upgrade-reboot.nu`? We could do a download-only first, then reboot without applying, verify it isn't applied etc